client/cli/src/commands/runcmd.rs

```diff
@@ -564,24 +564,30 @@ fn interface_str(
 
 #[derive(Debug)]
 enum TelemetryParsingError {
-	MissingVerbosity
+	MissingVerbosity,
+	VerbosityParsingError(std::num::ParseIntError),
+	UrlParsingError(std::convert::Infallible),
 }
 
 impl std::error::Error for TelemetryParsingError {}
 
 impl fmt::Display for TelemetryParsingError {
 	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "Verbosity level missing")
+		match &*self {
+			TelemetryParsingError::MissingVerbosity => write!(f, "Verbosity level missing"),
+			TelemetryParsingError::VerbosityParsingError(e) => write!(f, "{}", e),
+			TelemetryParsingError::UrlParsingError(e) => write!(f, "{}", e),
+		}
     }
 }
 
-fn parse_telemetry_endpoints(s: &str) -> Result<(String, u8), Box<dyn std::error::Error>> {
+fn parse_telemetry_endpoints(s: &str) -> Result<(String, u8), TelemetryParsingError> {
 	let pos = s.find(' ');
 	match pos {
-		None => Err(Box::new(TelemetryParsingError::MissingVerbosity)),
+		None => Err(TelemetryParsingError::MissingVerbosity),
 		Some(pos_) => {
-			let verbosity = s[pos_ + 1..].parse()?;
-			let url = s[..pos_].parse()?;
+			let verbosity = s[pos_ + 1..].parse().map_err(TelemetryParsingError::VerbosityParsingError)?;
+			let url = s[..pos_].parse().map_err(TelemetryParsingError::UrlParsingError)?;
```

Comment:
```suggestion
			let url = s[..pos_].to_string();
```

I have no idea why we're using `parse()` here.

Reply:
Very good point. Maybe the original idea was to trying and parse the string to make sure it's a valid url, and then the code author forgot to change it?

Reply:
Checking that this is a valid url, sounds like a good idea. However, this will require more than just calling `parse()` here :) You will need a crate a provides support for url parsing/checking.

Reply:
Indeed! I was thinking about using the [url](https://docs.rs/url/2.1.1/url/) crate, but I'm not sure what are the practices regarding adding crates... ?

Reply:
Maybe libp2p re-exports something you can use. @tomaka ?

Reply:
This can be either a libp2p `Multiaddr` or a URL.
We should add in libp2p-telemetry a function to parse this and returns a strongly-typed struct.